File: dsa/stack/Stack.java
package stack;
import java.util.*;

public class Stack{
    static int maxElements = 100;
    static int top;
    static int arr[] = new int[maxElements];
    
    boolean isEmpty(){
        return (top < 0);
    }
    
    Stack(){
        top = -1;
    }

    static void push(int x){
        if(top >= (maxElements - 1)) throw new ArithmeticException("Stack Overflow");
        
        arr[++top] = x;
        System.out.println(x + " pushed into stack");
    }

    static int peek(){
        if (top < 0) throw new ArithmeticException("Stack Underflow");
        return arr[top];
    }

    static void pop(){
        if (top < 0) throw new ArithmeticException("Stack Underflow");

        int x = arr[top--];
        System.out.println("\nElement popped.");
        return;
    }

    static void printStack(){
        for(int i = top;i>-1;i--) System.out.print(" "+ arr[i]);
        System.out.println();
    }

    public static void main(String[] args) {
        Stack s = new Stack();
        s.push(5);
        s.push(6);
        s.push(56);
        s.push(9);
        s.push(7);
        s.push(10);

        pop();
        pop();

        printStack();
    }
}

File: cpp/stack.cpp
#include <bits/stdc++.h>
using namespace std;

#define n 5

void swap(int num1,int num2)
{
    int tmp = num1;
    num1 = num2;
    num2 = tmp;
}

class myStack
{
    int *arr;
    int top;

public:
    myStack()
    {
        arr = new int[n];
        top = -1;
    }

    void push(int val)
    {
        if (top == n - 1)
        {
            cout << "Stack Overflow" << endl;
            return;
        }
        top++;
        arr[top] = val;
        return;
    }

    void pop()
    {
        if (top == -1)
        {
            cout << "Stack already empty" << endl;
            return;
        }
        top--;
    }

    void DisplayStack()
    {
        int i = 0;
        while (i <= top)
        {
            cout << arr[i] <<"  ";
            i++;
        }
        cout<<endl;
        return;
    }

    int peek()
    {
        if (top == -1)
        {
            cout << "Oops! Stack is empty" << endl;
        }

        return arr[top];
    }

    void IsEmpty()
    {
        if (top == -1)
        {
            cout << "Stack is empty" << endl;
        }

        else
        {
            cout << "Stack is not empty" << endl;
        }
    }

    void Reverse()
    {
        int terms = 0;
        while (terms < top)
            terms++;

        int m, tmp;
        if (terms % 2 == 0)
            m = terms / 2;

        else
            m = (terms - 1) / 2;

        for (int i = 0; i < m; i++)
        {
            swap(arr[0+i], arr[terms-i-1]);
        }

        for (int i = 0; i < terms; i++)
        {
            cout << arr[i] << ", ";
        }
        cout << endl;
    }

    void debug(int index)
    {
        cout<<arr[index]<<endl;
    }
};

int main(void)
{
    myStack st;
    st.push(1);  //0
    st.push(2);  //1
    st.push(3);  //2
    st.push(4);  //3
    st.push(5);  //3
    // st.push(0);

    // st.pop();
    st.DisplayStack();
    // st.IsEmpty();

    st.Reverse();

    // st.debug(3);

    return 0;
}